docs/Key-derivation.md

```diff
@@ -0,0 +1,96 @@
+# Key Derivation
+
+As an outcome of the previous activation steps, a single shared secret `KEY_MASTER_SECRET` is established for PowerAuth Client and PowerAuth Server. While additional shared secrets could be established by repeating the activation process, this may not be very handy in all situations, since the activation process is quite complex and not very user-friendly.
+
+For this reason, PowerAuth establishes the concept of derived keys. Each derived key is computed using the KDF algorithm (see "Implementation details" section for the definition):
+
+```java
+SecretKey KEY_DERIVED = KDF.derive(KEY_MASTER_SECRET, INDEX);
```

Comment:
Since crypto 3.0 the index is no longer a `long` index, we switched to `byte[]`:

```
    public byte[] getIndex() {
        return ByteBuffer.allocate(16).putLong(0L).putLong(index).array();
    }
```

Perhaps we should mention it?

Reply:
That's a technical detail. The "crypto" documentation should use "integer" as a type here.